app/scripts/lib/message-manager.js

```diff
@@ -94,18 +82,18 @@ module.exports = class MessageManager extends EventEmitter{
   }
 
   _updateMsg (msg) {
-    let messages = this.getMsgList()
-    let index = messages.findIndex((message) => message.id === msg.id)
+    let index = this.messages.findIndex((message) => message.id === msg.id)
     if (index !== -1) {
-      messages[index] = msg
+      this.messages[index] = msg
     }
-    this._saveMsgList(messages)
+    this._saveMsgList()
   }
 
-  _saveMsgList (msgList) {
+  _saveMsgList () {
+    const unapprovedMsgs = this.getUnapprovedMsgs()
+    const unapprovedMsgCount = Object.keys(unapprovedMsgs).length
```

Comment:
yeah that would work
Guess i didnt want to calculate the unaproved messages twice